code/modules/projectiles/guns/energy/tetra.dm

```diff
@@ -0,0 +1,46 @@
+/obj/item/gun/energy/tetra
+	name = "\"Tetra\" laser SMG"
+	desc = "A overengineered, complex laser SMG. Mounts both a visible-light laser emitter and a near-infrared laser emitter, allowing it to fire both regular, moderate-power beams and quieter, lower-power invisible beams.\
+	This variant has a basic optic fitted."
+	icon = 'icons/obj/guns/energy/tetra.dmi'
+	icon_state = "tetra"
+	w_class = ITEM_SIZE_BULKY
+	fire_sound = 'sound/weapons/Laser2.ogg'
+	fire_sound_silenced = 'sound/weapons/quietlaser2.ogg'
+	suitable_cell = /obj/item/cell/medium
+	can_dual = TRUE
+	charge_meter = TRUE
+	origin_tech = list(TECH_COMBAT = 5, TECH_POWER = 6)
+	slot_flags = SLOT_BELT
+	matter = list(MATERIAL_PLASTEEL = 18, MATERIAL_STEEL = 8, MATERIAL_PLASTIC = 8, MATERIAL_SILVER = 12, MATERIAL_GOLD = 2)
+	price_tag = 1500
+	silenced = FALSE
+	damage_multiplier = 0.6
+	recoil_buildup = 1
+	one_hand_penalty = 5
+	zoom_factor = 0.2
+	projectile_type = /obj/item/projectile/beam
+	suitable_cell = /obj/item/cell/medium
+	charge_cost = 25
```

Comment:
```suggestion
	charge_cost = 50
```
This is my only concern rest looks more or less fine

Reply:
i tested it with 40 charge cost initially, it was still significantly weaker than the sunrise (which it's balanced against) so i pushed it down to 25

50 is a bit much imo, it makes it about firestorm tier

Reply:
I have to agree with Trilby on this. 25 chargecost means you can end up with 32 shots on an 800 cell, one of the more common types. Double that for a 1600 cell.  for the default, 12 dam per shot and 20AP(for 32 per hit against mobs, or 12 per hit against players.). 1.5 fire delay. It costs some time, it IS a guild craft but i'd say it's relatively cheap for being so good. Consider raising the fire cost just a bit - it's okay to have it be a bit underpowered at introduction and raise it from there. You'll meet less resistance to buffing something over time than you will nerfing it from a state that's a bit "too good".

Reply:
32 hit against mobs is incorrect since overpenetration damage is based on mob armor

anyways, i'll pull it down to 40 (20 shots on a 800 cell)

Reply:
👌